packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -698,14 +765,16 @@ class HotRunner extends ResidentRunner {
       restartEvent = 'exception';
       return OperationResult(1, 'hot restart failed to complete: $err\n$st', fatal: true);
     } finally {
-      HotEvent(restartEvent,
-        targetPlatform: targetPlatform,
-        sdkName: sdkName,
-        emulator: emulator,
-        fullRestart: true,
-        reason: reason,
-        fastReassemble: null,
-      ).send();
+      if (restartEvent != null) {
```

Comment:
this shouldn't ever be null right?

Reply:
I added new fields when it succeed. In here, if hot restart succeeded, the event will be sent above, and restartEvent will be null. The code here is to handle failure cases only. Added comment to explain.

Reply:
Ahh, got it